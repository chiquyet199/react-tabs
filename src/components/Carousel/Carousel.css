/* Nuka Carousel style overrides */

.slider.light-dots .slider-control-bottomcenter ul li button {
  color: #ffffff !important;
}

.slider-control-bottomcenter ul {
  top: 30px !important;
}

.slider-control-bottomcenter ul li button {
  padding: 0 !important;
  height: 24px;
  width: 20px;
  line-height: 1;
}
