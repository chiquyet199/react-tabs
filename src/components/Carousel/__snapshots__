/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Carousel/__native_snapshots__