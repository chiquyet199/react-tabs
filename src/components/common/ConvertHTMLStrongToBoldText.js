import React from "react";
import {Text} from "react-native";
import styled from "styled-components/native";

export const StyledView = styled(Text)`
  display: inline;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
`;

export default ({htmlString}) => {
  const parsed = [];
  let cursor = 0;
  let searchingFor = "<strong>";
  while (cursor < htmlString.length) {
    let endPoint = htmlString.indexOf(searchingFor, cursor);
    endPoint = endPoint === -1 ? htmlString.length : endPoint;
    const token = htmlString.substring(cursor, endPoint);
    if (searchingFor === "<strong>") {
      parsed.push(<Text key={token}>{token}</Text>);
    } else {
      // Styled-components seems to ignore font-weight here
      parsed.push(
        <Text style={{fontWeight: "bold"}} key={token}>
          {token}
        </Text>,
      );
    }
    cursor = endPoint + searchingFor.length;
    searchingFor = searchingFor === "<strong>" ? "</strong>" : "<strong>";
  }

  return <StyledView>{parsed}</StyledView>;
};
