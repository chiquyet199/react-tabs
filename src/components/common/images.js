import styled from "styled-components/native";

export const CenterIconImage = styled.Image`
  width: 80px;
  height: 80px;
  margin-top: 20px;
  margin-bottom: 20px;
  align-self: center;
`;

export const CenterPageImage = styled.Image`
  width: 370px;
  height: 320px;
  margin-top: 20px;
`;
