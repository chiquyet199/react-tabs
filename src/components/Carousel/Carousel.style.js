import styled from "styled-components/native";
import {themed} from "../../utils/common";

export const Container = styled.View`
  background-color: transparent;
  padding-left: 0;
  padding-right: 0;
  width: 100%;
  flex-wrap: wrap;
  align-items: flex-start;
  max-width: 340px;
  flex-shrink: ${({fullHeight}) => (fullHeight ? 1 : 0)};
  flex-grow: ${({fullHeight}) => (fullHeight ? 1 : 0)};
  justify-content: center;
  margin-top: 15px;
  margin-bottom: 15px;
`;
export const SlideContainer = styled.View`
  justify-content: center;
  align-items: center;
`;
export const CurrentSlide = styled.View`
  margin-bottom: 14px;
  flex-shrink: 0;
  flex-grow: 0;
  justify-content: center;
  align-items: center;
`;
export const CurrentImage = styled.Image`
  width: ${({imageWidth}) => imageWidth || 340}px;
  height: ${({imageHeight}) => imageHeight || 190}px;
  border-radius: 0;
`;

export const TitleText = themed(styled.Text`
  width: 100%;
  color: ${({theme, globaltheme}) => theme[globaltheme].textColor};
  margin-top: 10px;
  margin-bottom: 14px;
  font-size: 26px;
  line-height: 30px;
  text-align: center;
`);

export const MoreText = themed(styled.Text`
  width: 100%;
  color: ${({theme, globaltheme}) => theme[globaltheme].textColor};
  font-size: 14px;
  line-height: 21px;
  text-align: center;
  margin-bottom: 38px;
`);
