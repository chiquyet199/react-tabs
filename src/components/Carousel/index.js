import React, {Component} from "react";
import Carousel, {Pagination} from "react-native-snap-carousel";
import {
  Container,
  CurrentSlide,
  CurrentImage,
  TitleText,
  MoreText,
} from "./Carousel.style";
import {themed} from "../../utils/common";
import {connect} from "react-redux";
import {CAROUSEL_MAX_WIDTH} from "../../constants/sizes";
import {
  any,
  arrayOf,
  bool,
  element,
  number,
  object,
  shape,
  string,
} from "prop-types";

// TODO add i18n

export class NativeCarousel extends Component {
  constructor(props) {
    super(props);
    this.props = props;
    this._carousel = {};
    this.state = {
      activeSlide: 0,
    };
  }

  get pagination() {
    const {activeSlide} = this.state;
    const {slides, dots, theme} = this.props;

    return (
      <Pagination
        dotsLength={slides.length}
        activeDotIndex={activeSlide}
        containerStyle={{
          position: "absolute",
          bottom: -10,
          left: 0,
          right: 0,
        }}
        dotStyle={{
          width: 8,
          height: 8,
          borderRadius: 5,
          margin: 0,
          paddingHorizontal: 0,
          backgroundColor: `${
            dots === "light" ? theme.colors.white : theme.colors.canopySteel
          }`,
        }}
        inactiveDotOpacity={0.4}
        inactiveDotScale={1}
      />
    );
  }

  handleSnapToItem = index => {
    this.setState({activeSlide: index});
  };

  renderItem = ({item, index, fullWidth, screenWidth}) => {
    const imageUrl = item.url || item.thumbnail;
    const isLocalImage = typeof imageUrl === "number";
    const itemWidth = fullWidth ? screenWidth : CAROUSEL_MAX_WIDTH;

    return (
      <CurrentSlide
        onPress={() => {
          this._carousel.snapToItem(index);
        }}
      >
        <CurrentImage
          imageHeight={200}
          imageWidth={itemWidth}
          source={isLocalImage ? imageUrl : {uri: imageUrl}}
        />
        {item.title && <TitleText>{item.title}</TitleText>}
        {item.text && <MoreText>{item.text}</MoreText>}
        {item.node}
      </CurrentSlide>
    );
  };

  render = () => {
    const {activeSlide} = this.state;
    const {
      slides,
      sliderWidth = CAROUSEL_MAX_WIDTH,
      itemWidth = CAROUSEL_MAX_WIDTH,
      fullWidth = false,
      fullHeight,
      screenWidth,
    } = this.props;

    return (
      <Container
        width={fullWidth ? "100%" : sliderWidth}
        fullHeight={fullHeight}
      >
        <Carousel
          ref={c => {
            this._carousel = c;
          }}
          autoplay
          loop
          autoplayInterval={6000}
          enableMomentum={false}
          lockScrollWhileSnapping
          data={slides}
          renderItem={this.renderItem}
          onSnapToItem={this.handleSnapToItem}
          inactiveSlideOpacity={0}
          sliderWidth={fullWidth ? screenWidth : sliderWidth}
          itemWidth={fullWidth ? screenWidth : itemWidth}
          layout="default"
          firstItem={activeSlide}
        />
        {this.pagination}
      </Container>
    );
  };
}

const mapStateToProps = state => ({
  screenWidth: state.dimensions.width,
});

const mapDispatchToProps = {};

NativeCarousel.propTypes = {
  dots: string,
  fullHeight: bool,
  fullWidth: bool,
  itemWidth: number,
  screenWidth: number,
  sliderWidth: number,
  slides: arrayOf(
    shape({
      title: string,
      text: string,
      url: any,
      thumbnail: any,
      node: element,
    }),
  ).isRequired,
  theme: object,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(themed(NativeCarousel));
