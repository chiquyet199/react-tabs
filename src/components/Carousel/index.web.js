import React, {Fragment} from "react";
import Carousel from "nuka-carousel";
import {TitleText, MoreText, Container, SlideContainer} from "./Carousel.style";
import {themed} from "../../utils/common";
import {any, arrayOf, bool, element, shape, string} from "prop-types";
import styles from "./Carousel.css";

export const WebCarousel = ({
  dots,
  fullHeight,
  fullWidth,
  slides,
  wrapAround,
}) => {
  return (
    <Container width={fullWidth && "100%"} fullHeight={fullHeight}>
      <Carousel
        className={dots === "light" && "light-dots"}
        width="100%"
        renderCenterLeftControls={() => null}
        renderCenterRightControls={() => null}
        autoplay
        autoPlayInterval={6000}
        wrapAround={wrapAround}
        pauseOnHover={false}
        initialSlideHeight={180}
        initialSlideWidth={340}
        dragging={false}
        {...styles}
      >
        {slides.map(slide => (
          <Fragment key={slide.thumbnail || slide.url}>
            <img src={slide.thumbnail || slide.url} alt={slide.title} />
            <SlideContainer>
              {slide.title && <TitleText>{slide.title}</TitleText>}
              {slide.text && <MoreText>{slide.text}</MoreText>}
              {slide.node}
            </SlideContainer>
          </Fragment>
        ))}
      </Carousel>
    </Container>
  );
};

WebCarousel.defaultProps = {
  wrapAround: true,
};

WebCarousel.propTypes = {
  dots: string,
  fullWidth: bool,
  fullHeight: bool,
  slides: arrayOf(
    shape({
      title: string,
      text: string,
      url: any,
      thumbnail: any,
      node: element,
    }),
  ).isRequired,
  wrapAround: bool,
};
export default themed(WebCarousel);
