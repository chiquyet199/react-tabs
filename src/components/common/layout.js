import React from "react";
import styled from "styled-components/native";
import {isNative, isWeb, responsive, themed} from "../../utils/common";
import {TouchableOpacity} from "react-native";
import {DESKTOP_CONTAINER_WIDTH} from "../../constants/sizes";
import {ButtonContainer} from "../Button/Button";
import {func, node, number, string} from "prop-types";

export const FlexContainer = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
`;

export const Flexed = styled.View``;

export const Flex = styled.View`
  margin-top: ${({theme}) => theme.padding.top + "px"};
  position: absolute;
  width: 100%;
  background-color: transparent;
`;

export const FlexColumn = styled.View`
  max-width: 537px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const TopBarContainer = styled.View`
  width: 100%;
  max-width: 100%;
  padding: 0;
  padding-top: ${({theme}) => theme.padding.top + "px"}
  z-index: 2;
  position: relative;
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].topbar.background}
`;

export const BaseContainer = styled.View`
  flex: 1;
  padding: 0;
  width: 100%;
  z-index: 1;
`;

export const CenteredContainer = styled.View`
  flex: 1;
  justify-content: ${({align}) => align || "center"};
  align-self: center;
  align-items: center;
  width: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
  margin-horizontal: auto;
  padding-horizontal: ${({full}) => (full === "true" ? "0" : "10px")};
  flex-basis: auto;
`;

export const EndPageContainer = responsive(CenteredContainer)({
  S: `justify-content: flex-start`,
});

export const PageContainer = responsive(styled(CenteredContainer)`
  width: 100%;
  max-width: ${({full}) =>
    full === "true" ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  align-self: center;
  align-items: stretch;
  margin-top: ${({layout}) => (layout === "hq" ? "60px" : "40px")};
  ${({margintop}) => margintop >= 0 && `margin-top: ${margintop}px`};
  padding-horizontal: 10px;
  justify-content: center;
  flex-basis: auto;
`)({
  S: `
      padding-top: ${({theme}) => theme.padding.top + "px"};
      justify-content: space-between;
      padding-horizontal: 10px;
      margin-top: ${({layout}) => (layout === "hq" ? "80px" : "20px")};
    `,
});

export const FormContainer = responsive(styled.View`
  justify-content: center;
  align-items: center;
  max-width: ${DESKTOP_CONTAINER_WIDTH}px;
  width: 100%;
  align-self: center;
`)({
  S: `
      padding-top: 10px;
      justify-content: space-between;
      max-width: 100%;
    `,
});

export const FlexRow = styled.View`
  flex: 1;
  flex-direction: row;
  flex-basis: 100%;
`;

export const ButtonBar = styled.View`
  max-width: ${({isMenu}) =>
    isMenu ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  width: 100%;
  border-top-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.lightSteel : theme.colors.none};
  border-top-width: ${({isMenu}) => (isMenu ? "1px" : 0)};
  background-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.white : theme.colors.none};
  flex-direction: row;
  margin-horizontal: 0;
  margin-bottom: ${({isMenu}) => (isMenu ? 0 : "5px")};
  flex-wrap: ${({wrap}) => (wrap ? "wrap" : "nowrap")};
  ${({mt}) => mt && `margin-top: ${mt}px}`};
  ${ButtonContainer} {
    margin: 20px;
  }
`;

export const BottomButtonBar = responsive(ButtonBar)({
  S: `flexGrow: 10;
      flexDirection: row;
      alignItems: center;
      justifyContent: flex-end;
    `,
});

export const ResponsiveWebScrollView = responsive(styled.ScrollView``)({
  M: `
        width: 500px;
  `,
  L: `
        width: 600px;
    `,
});

export const PageTitle = themed(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding-horizontal: 5px;
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  ${({mb}) => mb && `margin-bottom: ${mb}px}`};
  ${({color}) => color && `color: ${color}`};
`);

export const TextParagraph = themed(styled.Text`
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  color: ${({disabled, theme, globaltheme}) =>
    disabled
      ? theme[globaltheme].disabledTextColor
      : theme[globaltheme].baseColor};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding: ${({gutters}) => gutters || "15px 5px"};
  flex-shrink: 0;
  flex-grow: 0;
  ${({mb}) => mb && `margin-bottom: ${mb}px`};
  ${({color}) => color && `color: ${color}`};
`);

export const ViewParagraph = themed(styled.View`
  width: 100%;
  flex-shrink: 0;
  flex-grow: 0;
`);

export const SubTitle = themed(styled(TextParagraph)`
  font-weight: bold;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  ${({color}) => color && `color: ${color}`};
`);

export const TextLinkText = themed(styled.Text`
  text-align: center;
  font-size: 16;
  line-height: 24;
  color: ${({theme, globaltheme}) => theme[globaltheme].accentColor};
  font-weight: 700;
  padding-top: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")};
  padding-bottom: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-left: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-right: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  text-align:  ${({align}) => align || "center"};
`);

export const TextLink = ({onClick, children, gutters, align, globaltheme}) => (
  <TouchableOpacity onPress={onClick}>
    <TextLinkText align={align} gutters={gutters} globaltheme={globaltheme}>
      {children}
    </TextLinkText>
  </TouchableOpacity>
);

TextLink.propTypes = {
  onClick: func,
  children: node,
  gutters: number,
  align: string,
  globaltheme: string,
};

export const NotchSafety = styled.View`
  height: 40px;
`;

export const PageHeader = styled.Text`
  font-size: 18;
  text-align: center;
  width: 100%;
  margin-top: ${({theme}) => theme.padding.top + 15 + "px"};
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  position: absolute;
`;

export const HeadlineText = responsive(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  margin: 8px 0 40px;
  padding: 0;
`)({
  S: `
    padding-left: 15px;
    `,
  L: `
    align-self: center;
   `,
});

export const BoxButtonBar = styled.View`
  align-self: center;
  align-items: stretch;
  width: 100%;
  margin-horizontal: auto;
  flex-direction: row;
`;

export const FixedWidthContainer = styled.View`
  width: ${DESKTOP_CONTAINER_WIDTH}px;
  margin-horizontal: auto;
  padding-horizontal: 15px;
`;

export const PaddedContent = styled.View`
  padding-horizontal: 10px;
`;

export const ErrorMessage = themed(styled(TextParagraph)`
  color: ${({theme}) => theme.colors.alertRed};
`);

// Scroll layout

export const ScrollContentContainer = styled.View`
  width: 100%;
  ${isNative ? "flex: 1;" : "height: calc(100vh - 52px)"};
  flex-grow: ${() => (isWeb ? 0 : 1)};
  flex-shrink: ${() => (isWeb ? 0 : 1)};
  justify-content: center;
  align-self: center;
  align-items: center;
`;

export const ScrollButtonBar = styled(ButtonBar)`
  padding-horizontal: 10px;
  ${({mt}) => mt && `margin-top: ${mt}px}`};
`;

export const ScrollScrollView = styled.ScrollView`
  ${isNative ? "" : "height: 100%"};
  width: 100%;
  flex-grow: 1;
  flex-shrink: 1;
`;
