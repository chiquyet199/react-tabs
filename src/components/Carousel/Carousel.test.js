import React from "react";
import {View, Text} from "react-native";
import {shallow} from "enzyme";
import {extend} from "underscore";
import {NativeCarousel} from "./index";
import {WebCarousel} from "./index.web";
import {Container} from "./Carousel.style";
import {isNative} from "../../utils/common";
import ReactNativeSnapCarousel from "react-native-snap-carousel";

describe("Carousel - page tests", () => {
  it("matches snapshot", () => {
    const comp = isNative ? getNativeComponent() : getWebComponent();
    expect(comp).toMatchSnapshot();
  });

  it("renders with fullWidth prop ", () => {
    const screenWidth = 1000;
    if (isNative) {
      const comp = getNativeComponent({
        fullWidth: true,
        screenWidth,
      });
      expect(comp.find(Container).prop("width")).toBe("100%");
      expect(comp.find(ReactNativeSnapCarousel)).toHaveProp(
        "sliderWidth",
        screenWidth,
      );
    } else {
      const comp = getWebComponent({
        fullWidth: true,
        screenWidth,
      });
      expect(comp.find(Container).prop("width")).toBe("100%");
    }
  });

  const getWebComponent = props => {
    const parsedProps = extend(getWebMockProps(), props);

    return shallow(<WebCarousel {...parsedProps} />);
  };

  const getNativeComponent = props => {
    const parsedProps = extend(getNativeMockProps(), props);

    return shallow(<NativeCarousel {...parsedProps} />);
  };

  const getBaseProps = () => ({
    slides: [
      {
        title: "title1",
        thumbnail: "test.jpg",
        text: "text text",
        node: (
          <View>
            <Text>Lorem</Text>
            <Text>ipsum</Text>
          </View>
        ),
      },
      {
        title: "title2",
        thumbnail: "test2.jpg",
        text: "text text",
        node: <Text>Lorem ipsum</Text>,
      },
    ],
    dots: "light",
    theme: {colors: {white: "#fff", canopySteel: "#888"}},
  });

  const getNativeMockProps = () => {
    return extend(getBaseProps(), {sliderWidth: 500, itemWidth: 500});
  };

  const getWebMockProps = () => {
    return extend(getBaseProps(), {wrapAround: true});
  };
});
