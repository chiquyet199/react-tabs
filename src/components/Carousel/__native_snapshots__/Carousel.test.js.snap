// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel - page tests matches snapshot 1`] = `
<Styled(Component)
  width={500}
>
  <Carousel
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    autoplay={true}
    autoplayDelay={5000}
    autoplayInterval={6000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    data={
      Array [
        Object {
          "node": <Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Lorem
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              ipsum
            </Text>
          </Component>,
          "text": "text text",
          "thumbnail": "test.jpg",
          "title": "title1",
        },
        Object {
          "node": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Lorem ipsum
          </Text>,
          "text": "text text",
          "thumbnail": "test2.jpg",
          "title": "title2",
        },
      ]
    }
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    hasParallaxImages={false}
    inactiveSlideOpacity={0}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    itemWidth={500}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={true}
    loop={true}
    loopClonesPerSide={3}
    onSnapToItem={[Function]}
    renderItem={[Function]}
    scrollEnabled={true}
    shouldOptimizeUpdates={true}
    slideStyle={Object {}}
    sliderWidth={500}
    swipeThreshold={20}
    useScrollView={false}
    vertical={false}
  />
  <Pagination
    activeDotIndex={0}
    containerStyle={
      Object {
        "bottom": -10,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
    dotStyle={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": 8,
        "margin": 0,
        "paddingHorizontal": 0,
        "width": 8,
      }
    }
    dotsLength={2}
    inactiveDotOpacity={0.4}
    inactiveDotScale={1}
    tappableDots={false}
    vertical={false}
  />
</Styled(Component)>
`;
